nova/ci/database/DB_active_rec.php

```diff
@@ -1163,6 +1163,22 @@ public function set_insert_batch($key, $value = '', $escape = TRUE)
 	 */
 	function insert($table = '', $set = NULL)
 	{
+		$ci =& get_instance();
+		if($table && ($routerClass = $ci->router->fetch_class()) && ($routerMethod = $ci->router->fetch_method()))
+		{
+			$ci->event->trigger([
+				'db',
+				'insert',
+				'prepare',
```

Comment:
What are we doing with these 3 pieces of data exactly? Are there situations where we need all 3 or would omit certain ones?

Reply:
The reason for all of these different segments is to allow someone to listen at a particular point in time. Specifically for these, you might want to do `->event->listen(['db'], ...)` to attach logging for all database calls, or you might just want to listen for `->event->listen(['db', 'insert', 'prepare'], ...)` for some specific action you take whenever you're preparing data for an insert. I didn't define another event for DB insert - at this time - but I could see us maybe wanting to add an event after the insert occurs, hence why I used `prepare` instead of it just being `['db', 'insert']`.